version: '3.8'

services:
  web:
    build: .
    container_name: flask_app
    restart: always
    environment:
      - FLASK_DEBUG=False
      - MODEL_LOAD_NAME=NLLB-rup-ron-eng-ct2-v2
      - MAX_INPUT_TOKENS=1024
      - BATCH_SIZE=4
      - MAX_OUTPUT_TOKENS=1024
      - FLASK_PORT=5000
      - FLASK_HOST=0.0.0.0
    expose:
      - "5000"
    networks:
      - app-network

  nginx:
    image: nginx:latest
    container_name: nginx_proxy
    restart: always
    ports:
      - "80:80"
      - "443:443"
    volumes:
      - ./nginx.conf:/etc/nginx/nginx.conf
      - ./letsencrypt:/etc/letsencrypt  # Mount Let's Encrypt certificates
      - ./certbot/www:/var/www/certbot  # Certbot challenge
    depends_on:
      - web
    networks:
      - app-network

  certbot:
    image: certbot/certbot
    container_name: certbot
    volumes:
      - ./letsencrypt:/etc/letsencrypt  # Store certificates
      - ./certbot/www:/var/www/certbot  # Certbot challenge
    entrypoint: "/bin/sh -c 'trap exit TERM; while :; do sleep 6h & wait $${!}; certbot renew; done;'"
    networks:
      - app-network

networks:
  app-network:
    driver: bridge
